src/main/kotlin/io/sip3/salto/ce/sip/SipTransaction.kt

```diff
@@ -78,16 +99,28 @@ class SipTransaction {
                     caller = message.fromUserOrNumber()!!
                 }
 
-                when (message.statusCode) {
-                    100 -> if (tryingAt == null) tryingAt = packet.createdAt
-                    in 180..183 -> if (ringingAt == null) ringingAt = packet.createdAt
+                val statusCode = message.statusCode
```

Comment:
I don't like this piece of code with 2 `when` called one after each other. Plus we never go to the condition where 100 Trying will became `Proceeding`. It could be simplified. Another problem is that 183 and 200 may come not in a right order, so we always have to check if transaction was already terminated.